src/main/java/cn/nukkit/item/ItemMap.java

```diff
@@ -0,0 +1,111 @@
+package cn.nukkit.item;
+
+import cn.nukkit.Player;
+import cn.nukkit.block.Block;
+import cn.nukkit.level.Level;
+import cn.nukkit.nbt.tag.CompoundTag;
+import cn.nukkit.network.protocol.ClientboundMapItemDataPacket;
+import cn.nukkit.utils.MainLogger;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+/**
+ * Created by CreeperFace on 18.3.2017.
+ */
+public class ItemMap extends Item {
+
+    public static int mapCount = 0;
+
+    public ItemMap() {
+        this(0, 1);
+    }
+
+    public ItemMap(Integer meta) {
+        this(meta, 1);
+    }
+
+    public ItemMap(Integer meta, int count) {
+        super(MAP, 0, count, "Map");
+
+        if (!hasCompoundTag() || !getNamedTag().contains("map_uuid")) {
+            CompoundTag tag = new CompoundTag();
+            tag.putString("map_uuid", "" + mapCount++);
+            this.setNamedTag(tag);
+        }
+    }
+
+    public void setImage(BufferedImage img) {
+        try {
+            BufferedImage image = img;
+
+            if (img.getHeight() != 128 || img.getWidth() != 128) { //resize
+                image = new BufferedImage(128, 128, img.getType());
+                Graphics2D g = image.createGraphics();
+                g.drawImage(img, 0, 0, 128, 128, null);
+                g.dispose();
+            }
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ImageIO.write(image, "png", baos);
```

Comment:
You can add support and for video? Like MiNET?

And how add images with that? 📦